pallets/xtransfer/src/xcm/xcm_transfer.rs

```diff
@@ -237,6 +227,22 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		// Return dest chain and beneficiary on dest chain
+		pub fn extract_dest(dest: &MultiLocation) -> Option<(MultiLocation, MultiLocation)> {
```

Comment:
Do you think we can use `reanchor` to extract the dest properly?

https://github.com/paritytech/polkadot/blob/0a34022e31c85001f871bb4067b7d5f5cab91207/xcm/src/v1/multilocation.rs#L331

I think a proper way would be:

1. Extract the parachain location (only relaychain  ".." and "../para()" allowed)
2. Extract the beneficiary from `dest` by reanchor it to the parachain location

For example, for `../para(1000)/abc/def`, it will return `(../para(100), abc/def)`

Reply:
Haven't try it yet, looks like it may not work as expected. And for now actually i have no idea about splitting `dest` into `chain` part and `non-chain` part(as your suggestion) or just splitting last interior as beneficiary, because it's dest chain finally decide how to handle this. In common use cases, we give  a dest to other parachains like `(1, X2(Parachain(2000), AccountId32 {...}))`, it can work in both way.